unstructured_ingest/v2/processes/connectors/delta_table.py

```diff
@@ -0,0 +1,160 @@
+        default_factory=lambda: DeltaTableUploadStagerConfig()
+    )
+
+    def run(
+        self,
+        elements_filepath: Path,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        from unstructured_ingest.utils.table import convert_to_pandas_dataframe
+
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.parquet")
+
+        df = convert_to_pandas_dataframe(elements_dict=elements_contents)
+        df.to_parquet(output_path)
+
+        return output_path
+
+
+class DeltaTableUploaderConfig(UploaderConfig):
+    pass
+
+
+@dataclass
+class DeltaTableUploader(Uploader):
+    upload_config: DeltaTableUploaderConfig
+    connection_config: DeltaTableConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    @requires_dependencies(["deltalake"], extras="delta-table")
+    def precheck(self):
+        pass
+
+    def process_csv(self, csv_paths: list[Path]) -> pd.DataFrame:
+        logger.debug(f"uploading content from {len(csv_paths)} csv files")
+        df = pd.concat((pd.read_csv(path) for path in csv_paths), ignore_index=True)
+        return df
+
+    def process_json(self, json_paths: list[Path]) -> pd.DataFrame:
+        logger.debug(f"uploading content from {len(json_paths)} json files")
+        all_records = []
+        for p in json_paths:
+            with open(p) as json_file:
+                all_records.extend(json.load(json_file))
+
+        return pd.DataFrame(data=all_records)
+
+    def process_parquet(self, parquet_paths: list[Path]) -> pd.DataFrame:
+        logger.debug(f"uploading content from {len(parquet_paths)} parquet files")
+        df = pd.concat((pd.read_parquet(path) for path in parquet_paths), ignore_index=True)
+        return df
+
+    def read_dataframe(self, path: Path) -> pd.DataFrame:
+        if path.suffix == ".csv":
+            return self.process_csv(csv_paths=[path])
+        elif path.suffix == ".json":
+            return self.process_json(json_paths=[path])
+        elif path.suffix == ".parquet":
+            return self.process_parquet(parquet_paths=[path])
+        else:
+            raise ValueError(f"Unsupported file type, must be parquet, json or csv file: {path}")
+
+    @requires_dependencies(["deltalake"], extras="delta-table")
+    def run(self, path: Path, file_data: FileData, **kwargs: Any) -> None:
+        from deltalake.writer import write_deltalake
+
+        df = self.read_dataframe(path)
+        logger.info(
+            f"writing {len(df)} rows to destination table "
+            f"at {self.connection_config.table_uri}\ndtypes: {df.dtypes}",
+        )
+        storage_options = {}
+        secrets = self.connection_config.access_config.get_secret_value()
+        if secrets.aws_region and secrets.aws_access_key_id and secrets.aws_secret_access_key:
```

Comment:
Since this is purely based on values from the connection config, let's have a method on that that updates the existing storage options with all of these s3 ones if the values are present:
```python
storage_options = {}
self.connection_config.update_storage_options(storage_options=storage_options)
```